### PR TITLE
Add check to see if teacher before creating course

### PR DESCRIPTION
**Pull request by yannick-mayeur**

### CODE REVIEW

File: src/routes/course.route.js [hunk 1]
```diff
@@ -15,14 +15,20 @@ module.exports = (router) => {
...
-  router.post('/courses', async (req, res) => {
+  router.post('/courses', login, async (req, res) => {
     logger.log('info', 'received request: POST /courses\nbody:', req.body);
-    M.Course.create(req.body)
-      .then((course) => res.status(200).send(course))
-      .catch((err) => {
-        logger.log('error', 'POST /courses failed', err);
-        res.sendStatus(500);
-      });
+    if (req.user.role == 1) {
```
Comment (Alexis559) [inline src/routes/course.route.js]: Maybe you should use the '===' instead of '=='.

File: src/routes/course.route.js [hunk 2]
```diff
@@ -15,14 +15,20 @@ module.exports = (router) => {
...
+      M.Course.create(req.body)
+        .then((course) => res.status(200).send(course))
+        .catch((err) => {
+          logger.log('error', 'POST /courses failed', err);
+          res.sendStatus(500);
+        });
+    } else {
+      logger.log('warning', 'Not a teacher trying to POST /courses');
+      res.statusMessage = 'You have to be a teacher to create a course.';
+      res.sendStatus(500);
```
Comment (Alexis559) [inline src/routes/course.route.js]: Here I think the response code is not the most appropriate maybe 403 could be better.

**Comment by yannick-mayeur** [comment]
From issue #13

**Comment by yannick-mayeur** [comment]
Redone by @Alexis559

**Pull request closed without merging**